File: 142-linked-list-cycle-ii/linked-list-cycle-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def detectCycle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        crr=head
        seen=set()
        while crr!=None:
            if crr in seen:
                return crr
            seen.add(crr)
            crr=crr.next
        return None

File: 142-linked-list-cycle-ii/linked-list-cycle-ii.java

public class Solution {
    public ListNode detectCycle(ListNode head) {
        HashSet<ListNode> seen=new HashSet<>();
        ListNode crr=head;
        while(crr!=null){
            if(seen.contains(crr)==true){
                return crr;
            }
            seen.add(crr);
            crr=crr.next;
        }
        return null;
    }
}